core/lib/dal/src/proof_generation_dal.rs

```diff
@@ -88,6 +88,48 @@ impl ProofGenerationDal<'_, '_> {
         Ok(result)
     }
 
+    pub async fn get_available_batch(&mut self) -> DalResult<Option<L1BatchNumber>> {
+        let result: Option<L1BatchNumber> = sqlx::query!(
+            r#"
+            UPDATE proof_generation_details
+            SET
+                status = 'picked_by_prover',
+                updated_at = NOW(),
+                prover_taken_at = NOW()
+            WHERE
```

Comment:
This query locks the batch so that it cannot be picked by anyone else. I think we need a query that doesn't mutate state.